### PR TITLE
Converted like functionality to ajax request

### PR DESCRIPTION
**Pull request by PeterDMadrid**
**OVERVIEW**

- converted back like functionality to like and fix bugs

### CODE REVIEW

File: src/app/Http/Controllers/PostController.php [hunk 1]
```diff
@@ -62,19 +62,33 @@ public function viewPost(Post $post): View
     /**
      * Like Post
      */
-    public function like(Post $post): RedirectResponse
+    public function like(Post $post)
```
Comment (hipe01mark) [inline src/app/Http/Controllers/PostController.php]: return type!!!

File: src/app/Http/Controllers/PostController.php [hunk 2]
```diff
@@ -62,19 +62,33 @@ public function viewPost(Post $post): View
...
+        ]);
     }
-
+    
     /**
-     * Unlike Post
+     * UnLike Post
      */
-    public function unlike(Post $post): RedirectResponse
+    public function unlike(Post $post)
```
Comment (hipe01mark) [inline src/app/Http/Controllers/PostController.php]: return type!!!

File: src/app/Http/Controllers/PostController.php [hunk 3]
```diff
@@ -62,19 +62,33 @@ public function viewPost(Post $post): View
     /**
      * Like Post
      */
-    public function like(Post $post): RedirectResponse
+    public function like(Post $post)
     {
         $this->postService->like($post);
-        return redirect()->back();
+        $likeUsers = $post->likes->take(10)->map(function ($like) {
```
Comment (mikco-hipe) [inline src/app/Http/Controllers/PostController.php]: convert 10 to constant

File: src/app/Http/Controllers/PostController.php [hunk 4]
```diff
@@ -62,19 +62,33 @@ public function viewPost(Post $post): View
...
+    
     /**
-     * Unlike Post
+     * UnLike Post
      */
     public function unlike(Post $post): RedirectResponse
     {
         $this->postService->unlike($post);
-        return redirect()->back();
+        $likeUsers = $post->likes->take(10)->map(function ($like) {
```
Comment (mikco-hipe) [inline src/app/Http/Controllers/PostController.php]: convert 10 to constant

**Review by mikco-hipe — COMMENTED** [review]
Please check!

File: src/app/Http/Controllers/PostController.php (hunk 3 above)
Comment (PeterDMadrid) [reply src/app/Http/Controllers/PostController.php]: Created a constant variable inside the class

File: src/app/Http/Controllers/PostController.php (hunk 4 above)
Comment (PeterDMadrid) [reply src/app/Http/Controllers/PostController.php]: Done

File: src/app/Http/Controllers/PostController.php [hunk 5]
```diff
@@ -62,19 +61,33 @@ public function viewPost(Post $post): View
...
+        ]);
     }
-
+    
     /**
-     * Unlike Post
+     * UnLike Post
      */
-    public function unlike(Post $post): RedirectResponse
+    public function unlike(Post $post): JsonResponse
```
Comment (mikco-hipe) [inline src/app/Http/Controllers/PostController.php:80]: I dont think this returns JsonReponse alone, line 84 you are returning `$like->user->username;`

File: src/public/js/like-ajax.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                });
+
+                if (likesCount > 10) {
+                    const moreElement = document.createElement('div');
+                    moreElement.textContent = `and ${likesCount - 10} more...`;
+                    likeUsersElement.appendChild(moreElement);
+                }
+            }
+
+            // Toggle the like button class and text
```
Comment (mikco-hipe) [inline src/public/js/like-ajax.js]: removce comment

**Review by mikco-hipe — COMMENTED** [review]
Please check!

File: src/app/Http/Controllers/PostController.php (hunk 5 above)
Comment (mikco-hipe) [reply src/app/Http/Controllers/PostController.php:80]: joke tama pala

Comment (PeterDMadrid) [reply src/app/Http/Controllers/PostController.php:80]: The return $like->user->username; is inside the map function which then transforms the value of $post->likes. This then transferred into a variable that is returned using json response

**Review by mikco-hipe — APPROVED** [review]
LIGET 🙇

**Pull request merged**